pkg/k8shandler/common.go

```diff
@@ -382,146 +382,26 @@ func newPodTemplateSpec(nodeName, clusterName, namespace string, node api.Elasti
-
-	if commonRequestMem.IsZero() && commonLimitMem.IsZero() {
-		// no common memory settings
-		if nodeRequestMem.IsZero() && nodeLimitMem.IsZero() {
-			// no node settings, use defaults
-			lMem, _ := resource.ParseQuantity(defaultMemoryLimit)
-			limitMem = &lMem
-
-			rMem, _ := resource.ParseQuantity(defaultMemoryRequest)
-			requestMem = &rMem
-		} else {
-			// either one is not zero or both aren't zero but common is empty
-			if nodeRequestMem.IsZero() {
-				// request is zero use limit for both
-				requestMem = nodeLimitMem
-				limitMem = nodeLimitMem
-			} else {
-				if nodeLimitMem.IsZero() {
-					// limit is zero use request for both
-					requestMem = nodeRequestMem
-					limitMem = nodeRequestMem
-				} else {
-					// both aren't zero
-					requestMem = nodeRequestMem
-					limitMem = nodeLimitMem
-				}
-			}
-		}
-	} else {
-		// either one is not zero or both aren't zero (common)
-
-		//check node for override
-		if nodeRequestMem.IsZero() {
-			// no node request mem, check that common has it
-			if commonRequestMem.IsZero() {
-				requestMem = commonLimitMem
-			} else {
-				requestMem = commonRequestMem
-			}
-		} else {
-			requestMem = nodeRequestMem
-		}
+func newResourceRequirements(nodeResRequirements, commonResRequirements *v1.ResourceRequirements) *v1.ResourceRequirements {
 
-		if nodeLimitMem.IsZero() {
-			// no node request mem, check that common has it
-			if commonLimitMem.IsZero() {
-				limitMem = commonRequestMem
-			} else {
-				limitMem = commonLimitMem
-			}
-		} else {
-			limitMem = nodeLimitMem
-		}
-	}
-
-	// CPU
-	nodeLimitCPU := nodeResRequirements.Limits.Cpu()
-	nodeRequestCPU := nodeResRequirements.Requests.Cpu()
-	commonLimitCPU := commonResRequirements.Limits.Cpu()
-	commonRequestCPU := commonResRequirements.Requests.Cpu()
-
-	if commonRequestCPU.IsZero() && commonLimitCPU.IsZero() {
-		// no common memory settings
-		if nodeRequestCPU.IsZero() && nodeLimitCPU.IsZero() {
-			// no node settings, use defaults
-			lCPU, _ := resource.ParseQuantity(defaultCPULimit)
-			limitCPU = &lCPU
-
-			rCPU, _ := resource.ParseQuantity(defaultCPURequest)
-			requestCPU = &rCPU
-		} else {
-			// either one is not zero or both aren't zero but common is empty
-			if nodeRequestCPU.IsZero() {
-				// request is zero use limit for both
-				requestCPU = nodeLimitCPU
-				limitCPU = nodeLimitCPU
-			} else {
-				if nodeLimitCPU.IsZero() {
-					// limit is zero use request for both
-					requestCPU = nodeRequestCPU
-					limitCPU = nodeRequestCPU
-				} else {
-					// both aren't zero
-					requestCPU = nodeRequestCPU
-					limitCPU = nodeLimitCPU
-				}
-			}
-		}
-	} else {
-		// either one is not zero or both aren't zero (common)
-
-		//check node for override
-		if nodeRequestCPU.IsZero() {
-			// no node request mem, check that common has it
-			if commonRequestCPU.IsZero() {
-				requestCPU = commonLimitCPU
-			} else {
-				requestCPU = commonRequestCPU
-			}
+	resourceReq := nodeResRequirements
```

Comment:
one thing to note, since we pass this in as a pointer when we update `resourceReq` later on in this function it also updates `nodeResRequirements`. Not sure if this can cause unintended changes... do our unit tests cover this? Or can we update to not pass in a pointer?

Reply:
Good catch! This would appear in the CR when status.go does its updates